From c7398d4ecc5d81a4d916a234d5caa1774347acb9 Mon Sep 17 00:00:00 2001
From: Ole Streicher <olebole@debian.org>
Date: Tue, 23 May 2017 08:46:21 +0200
Subject: Consequently add -m32 flags if compiling for linux(32)

This allows cross compilation on a 64-bit Linux system with 32-bit
Linux as target system.

This requires to have according compiler support and libraries installed
(gcc-multilib and ia32-libs on older Debian/Ubuntu versions).
---
 unix/f2c/libf2c/makefile.u | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/unix/f2c/libf2c/makefile.u b/unix/f2c/libf2c/makefile.u
index 33d2e9c10..ccafa9df7 100644
--- a/unix/f2c/libf2c/makefile.u
+++ b/unix/f2c/libf2c/makefile.u
@@ -16,12 +16,12 @@
 CC ?= cc
 SHELL = /bin/sh
 CFLAGS = -O -w $(HSI_CF)
+LDFLAGS = $(HSI_LF)
 
 # compile, then strip unnecessary symbols
 .c.o:
 	$(CC) -c -DSkip_f2c_Undefs $(CFLAGS) $*.c
-	ld -r -x -o $*.xxx $*.o
-	mv $*.xxx $*.o
+
 ## Under Solaris (and other systems that do not understand ld -x),
 ## omit -x in the ld line above.
 ## If your system does not have the ld command, comment out
@@ -86,15 +86,15 @@ libf2c.a: $(OFILES)
 ## arrange for $DYLD_LIBRARY_PATH to include the directory containing libf2c.so.
 
 libf2c.so: $(OFILES)
-	$(CC) -shared -o libf2c.so $(OFILES)
+	$(CC) $(LDFLAGS) -shared -o libf2c.so $(OFILES)
 
 ### If your system lacks ranlib, you don't need it; see README.
 
 f77vers.o: f77vers.c
-	$(CC) -c f77vers.c
+	$(CC) $(CFLAGS) -c f77vers.c
 
 i77vers.o: i77vers.c
-	$(CC) -c i77vers.c
+	$(CC) $(CFLAGS) -c i77vers.c
 
 # To get an "f2c.h" for use with "f2c -C++", first "make hadd"
 hadd: f2c.h0 f2ch.add
-- 
2.39.2

